pkg/apis/cartographer/v1alpha1/workload_helpers.go

```diff
@@ -348,13 +349,13 @@ func (w *WorkloadSpec) MergeGit(git GitSource) {
 }
 
 func (w *WorkloadSpec) MergeSourceImage(image string) {
-	stash := w.Source
+	stash := w.Source.DeepCopy()
 	w.ResetSource()
 
 	w.Source = &Source{
 		Image: image,
 	}
-	if stash != nil && stash.Image != "" {
+	if stash != nil && stash.Subpath != "" {
```

Comment:
Why did this change? I want to understand the logic behind it. What happens if subpath comes as empty because user wants to remove it? Would it be deleted?

Reply:
There is no need to validate `stash.Subpath != ""` as after `w.ResetSource()` the sub-path will be `""`.
The logic here is that calling `w.ResetSource()` then `Source` become `nil`, then the `Image` is updated but the subpath is not so it needs to be updated as well.